src/util/get-github-client-config.ts

```diff
@@ -29,33 +31,34 @@ export async function getGitHubApiUrl(jiraHost: string, gitHubAppId: number) {
 const getGitHubClientConfigFromGitHubInstallationId = async (gitHubInstallationId: number): Promise<GitHubClientConfig> => {
 	const subscription = await Subscription.findOneForGitHubInstallationId(gitHubInstallationId);
 	const gitHubAppId = subscription?.gitHubAppId;
-	return getGitHubClientConfigFromAppId(gitHubAppId);
+	return getGitHubClientConfigFromAppId(gitHubAppId, subscription?.jiraHost);
 };
 
-const getGitHubClientConfigFromAppId = async (gitHubAppId: number | undefined): Promise<GitHubClientConfig> => {
+const getGitHubClientConfigFromAppId = async (gitHubAppId: number | undefined, jiraHost?: string): Promise<GitHubClientConfig> => {
 	const gitHubServerApp = gitHubAppId && await GitHubServerApp.getForGitHubServerAppId(gitHubAppId);
-	const gitHubCloudUrls = {
+	if (gitHubServerApp) {
+		return	{
+			hostname: gitHubServerApp.gitHubBaseUrl,
+			baseUrl: gitHubServerApp.gitHubBaseUrl,
+			appId: gitHubServerApp.appId,
+			privateKey: await gitHubServerApp.decrypt("privateKey")
```

Comment:
wouldn't it make sense to use keylocator here?

Reply:
We can but just want to save DB call.